python/cuml/dask/ensemble/base.py

```diff
@@ -109,15 +110,17 @@ def _fit(self, model, dataset, convert_dtype):
         else:
             self.num_classes = \
                 len(dask.array.unique(labels).compute())
-        labels = self.client.persist(dataset[1])
+
+        full_data = list(map(lambda x: x[1], data.gpu_futures))

```

Comment:
I know this shouldn't do any real work by itself as it's lazy, but maybe still do `combined_data = ... if broadcast else None` or something to ensure no work is done in the non-broadcast case just to make it maximally robust